components/org.wso2.carbon.identity.oauth.uma.permission.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/permission/endpoint/PermissionApiServiceImpl.java

```diff
@@ -215,9 +216,10 @@ private void handleErrorResponse(String message, UMAException umaException) thro
             if (log.isDebugEnabled()) {
                 log.debug(message, umaException);
             }
-            if (HandleErrorResponseConstants.RESPONSE_DATA_MAP.containsKey(code)) {
-                String statusCode = HandleErrorResponseConstants.RESPONSE_DATA_MAP.get(code)[0];
-                errorCode = HandleErrorResponseConstants.RESPONSE_DATA_MAP.get(code)[1];
+            Map<String, String[]> responseDataMap = HandleErrorResponseConstants.initializeResponseDataMap();
+            if (responseDataMap.containsKey(code)) {
```

Comment:
How do handle the case if the code does not exist?

Reply:
If the code doesn't exist it will be treated as a server error. All client error codes mentioned in UMA 2.0 specification are defined at [1].

[1] https://github.com/wso2-extensions/identity-oauth-uma/blob/master/components/org.wso2.carbon.identity.oauth.uma.common/src/main/java/org/wso2/carbon/identity/oauth/uma/common/HandleErrorResponseConstants.java#L29